### DISCUSSION TITLE
The 2-step process won't work for the use case I have for this tool. I need an alternative.

### DISCUSSION THREAD

**Discussion by BeverlyBrown** [rickysarraf/apt-offline#231] [General]
I read your latest wiki where it says to do set --update, apply the results and then do a set --upgrade.

That will not work for my use-case. I can only create the .sig file once and need to download all the packages I need for that and be able to install them on the offline machine.

I have a set of VMs deployed from the same OVA file. I need to upgrade it to match a newer version of the VM. I want to create the .sig on one of the older VMs, go to the newer VM to get the bundle of packages and apply that to all of the VMs instead of having  to get a .sig from each one. Doing a roundtrip that you described of creating the update .sig, getting the data, applying it, creating an upgrade .sig and then applying that will not work well for this case.

If I go ahead and try to use --update and --upgrade together so I can just cp the bundle to each of the older VMs and apply it, what problems will I run into. Is there some other way to mitigate or remediate any problems after the fact?

**Comment by rickysarraf** [comment]
It will still work with the exception that you'll have to wait for some upgrades in the next round-trip.


```
 ॐ ♅ ♄ ⛢  130  $  sudo apt-offline set /tmp/foo.uris --update --upgrade
Gathering details needed for 'update' operation
Gathering details needed for 'upgrade' operation
 13:26:23  rrs  var  tmp  Debian-Build  Result  
 ॐ ♅ ♄ ⛢  $  
```

Since yours' is a power-user case, you will have to tackle the differences and postpone them for the next iteration of the round-trip.

The recommendation is for `general users`
